Adjust settings to match RiverSR and general optical theory

steeleb opened the pull request:
Hey Katie!

Lots of little changes in here. Sorry to be 'that guy'. The meaningful changes are in `data_acquisition/src/gee_functions.py` and `data_acquisition/src/runGEEperTile.py`.

### masking hillshadow, adding hillshade rossyndicate/Landsat_C2_SRST#15
See the added `.updateMask(hs.eq(1))` on lines 374, 477, 518, 589 in `data_acquisition/src/gee_functions.py`. This change closes rossyndicate/Landsat_C2_SRST#15. I also added in those hillshade and shadow statistics to the reducer for our future Loire pull. The hillshadow mask is the only edit that is in this portion of the workflow that came out of our meeting for RiverSR/LakeSR.

### adding chebyshev centers [fka deepest points] rossyndicate/Landsat_C2_SRST#2 
This is long awaited - this mostly happens in lines 71-80 and 438-588 of `data_acquisition/src/runGEEperTile.py` and closes rossyndicate/Landsat_C2_SRST#2. 

Almost all other edits in here are fixing bugs that I've found while changing some of the yml settings. I created a new issue (#20), so that I can try to make sure that we have examples of all the options for internal testing. These will likely live in the `example_yml` folder.

### Other Notes
This is progress on rossyndicate/Landsat_C2_SRST#5 - I removed the in/out files from my messing around, 'setting' this up for a release. Note, the WRS2_descending shapefile stays in the `in` folder.

steeleb committed:
add hillshadow mask and add ee.filter not null to drop scenes that have no summary data

steeleb committed:
update some targets so that locs files runs

steeleb committed:
move in-house yamls to example folder

steeleb committed:
Merge branch 'rossyndicate:main' into main

steeleb committed:
remove redundnant runGEE file, now that we do it in PR branches

steeleb committed:
add ross wrs2, add list output of ls8/9 for future us

steeleb committed:
update to calculate centers

steeleb committed:
Merge branch 'main' of github.com:steeleb/Landsat_C2_SRST

steeleb committed:
fix centers calc and extraction

steeleb committed:
remove out files from repo

steeleb committed:
remove user files in in folder

steeleb committed:
move files around for repo

steeleb committed:
do not track files in the in/out folders

kathryn-willi reviewed (approved):
Looks great B... I may want to pick your brain about folder structure for this repo - but that's beyond this PR//more for my own understanding. Sorry this took so long!

steeleb committed:
add text to explain cases of errors

steeleb committed:
add notes about folder structure

steeleb committed:
add license blurb and contact info

steeleb commented:
Based on your feedback, I added more text to the run_targets.Rmd file about errored targets and added a folder architecture overview to the repository's README. 

I'm going to merge these changes, but I'd love to know if these added text bits help with the two items you commented on!

Thank you Katie!

kathryn-willi commented on run_targets.Rmd:
```diff
@@ -71,12 +76,19 @@ library(targets)
 
-Depending on how your config file is set up, you may have a few red circles on this graph. The important part is that the square on the right hand side is either orange or green.
+Depending on how your config file is set up, you may have a few red circles (errors) on this graph. 
+Errors for `polygons`, `centers`, and `locs` are okay in the pipeline, they are by design: these 
```
This text helps a lot!

steeleb replied:
YAY! I find it really helpful to know what isn't clear, because in my head it's crystal - and I might think I wrote out explanations... but in this case they are all in the _targets.R file that most end users would never look at. lol. Thanks for keeping me accountable.
